### DISCUSSION TITLE
Question about new file dialog

### DISCUSSION THREAD

**Discussion by quickfold** [freeplane/freeplane#2029] [General]
I'm going to hire someone to create a way to use the native Windows file open/save dialog in Freeplane. I'd like to donate this function to the open-source version if I can do so without adding significant cost. Even though it will only work for Windows, it can still help many Freeplane users.

@dpolivaev, are you interested in having this option available in Freeplane, even though it will only be for Windows OS? If yes, do you have any ideas on how you would like it integrated into Freeplane? Is there anything I should know in advance about the approach or doing PRs in Freeplane that would help me determine if it can be added to the official Freeplane without significant added cost? Thanks.

**Comment by dpolivaev** [comment]
I know that using native file dialogs has some advantages, but I have reasons why I have not integrated them. They do not allow to switch file types using drop-down box, and they do not allow to add a preview component. The drop-down box is currently used for switching the export type in the export dialog, for switching between template and user file directories in the open file dialog, and for selecting images of a certain type. The preview is used in image and in template selection. If we replace the currently used dialogs by native dialogs, we also need to find alternative solutions, and I have not got any ideas how to do it.

So, if you can come with a solution or even just suggest a solution, let me know.

**Reply by quickfold** [reply]
Apologies, my intent was not clear. I'm not suggesting that you _replace_ the Freeplane dialog to a native one, only that you allow Windows users an _option_ to use the native Windows dialog. There are advantages to the current file dialog and advantages to the native one. The advantages to the current file dialog are not very helpful to me, but the drawbacks are significant, so I want to create an option so I can have the advantages of the native one.

If you are not opposed to offering an option to others, and you would want me to have an option created in a way that would match your ideas about how to offer the option (such as ways you recommend or discourage doing it), please let me know. Otherwise, only I will be able to use it.

**Reply by dpolivaev** [reply]
I thought I answered. Sorry. Anyway: sure: if the option were available, I would happily include it.

**Reply by quickfold** [reply]
Thanks for answering. Can you tell me if you have any views on how it should be implemented for you to include it?

**Comment by quickfold** [comment]
@dpolivaev, I assume you support adding an option to Freeplane to allow each user to use it in the way most efficient for them. There are plusses and minuses to each approach, and neither is inherently better for every use case. I would like to know how you would prefer this option implemented to make it easiest to offer this option to others. If you do not want me to move forward on this option because you do not want Freeplane to let others choose to use a native dialog, please let me know so I can just move forward on my own.
